src/java.base/share/classes/java/security/interfaces/DSAPrivateKey.java

```diff
@@ -62,4 +62,20 @@ public interface DSAPrivateKey extends DSAKey, java.security.PrivateKey {
      * @return the value of the private key, {@code x}.
      */
     BigInteger getX();
+
+    /**
+     * Returns the parameters associated with this key.
+     * The parameters are optional and may be either
+     * explicitly specified or implicitly created during
+     * key pair generation.
+     *
+     * @implSpec
+     * The default implementation returns {@code null}.
+     *
+     * @return the associated parameters, may be {@code null}
+     * @since 22
+     */
+    default DSAParams getParams() {
```

Comment:
Suggestion: for this methods I recommend `@Override`-ing them and for the javadoc to used directed inheritance, roughly:

```
/**
 * {@inhertiDoc AsymmetricKey}
 * @implSpec ..
 * @return {@inhertiDoc AsymmetricKey}
 * @since 22
 */
```